packages/wonder-stuff-i18n/src/utils/extract-i18n.js

```diff
@@ -0,0 +1,508 @@
+
+/**
+ * Extract string segments and returns a unified string.
+ *
+ * This looks for individual string segments, like:
+ *     "hello" + " world"
+ *
+ * And returns a single merged string: hello world
+ */
+const extractAndCombineString = (textString: string): string => {
+    // Look for the string segments (either double or single quote)
+    const parts = [];
+    for (const match of textString.matchAll(INNER_STRINGS_RE)) {
+        parts.push(match[1] || match[2]);
+    }
+    // Merge them all together and convert the escaped characters
+    return convertEscapes(parts.join(""));
+};
+
+/**
+ * Extract a comment and returns a list of unified comment strings.
+ *
+ * This looks for comments that've potentially split across multiple lines
+ * and combines those segments together into a single comment string. It also
+ * handles multiple distinct comments in the same block and also removing
+ * duplicate comments.
+ *
+ * For example this is able to handle comments like this:
+ *
+ *     input: test
+ *     output: ["test]
+ *
+ *     input: hello\n// world
+ *     output: ["hello world"]
+ *
+ *     input: comment 1\n// I18N: comment 2
+ *     output: ["comment 1", "comment 2"]
+ *
+ *     input: comment 1\n// I18N: comment 1
+ *     output: ["comment 1"]
+ */
+const extractAndCombineComment = (commentString: string): Array<string> => {
+    const comments = [];
+
+    // Split the comment using the comment splitting regex above
+    for (const untrimmedComment of commentString.split(COMMENT_SPLIT_RE)) {
+        const comment = untrimmedComment.replace(/^[\s\n]*|[\s\n]*$/g, "");
+        // Are we dealing with a new comment?
+        if (comment.includes("I18N:")) {
+            comments.push(comment.replace(/I18N:\s*/, "").trim());
+        } else if (comment.trim()) {
+            // Or is this just appending to an existing comment?
+            const oldComment = comments[comments.length - 1];
+            comments[comments.length - 1] =
+                (oldComment ? oldComment + " " : "") + comment.trim();
+        }
+    }
+
+    // Get rid of all duplicate comments
+    return Array.from(new Set(comments));
+};
+
+/**
+ * Extract all i18n string (and comments) from a file.
+ *
+ * This method yields an object with the relevant details, namely:
+ *
+ *  // The line number of the string in the original file.
+ *  linePos: number,
+ *  // The starting position of the string in the original file.
+ *  startOffset: number,
+ *  // The ending position of the string in the original file.
+ *  endOffset: number,
+ *  // The type of string, such as "_", "$_", or "ngettext".
+ *  type: string,
+ *  // The strings found (will be one for singular, two for plural)
+ *  msgids: Array<string>,
+ *  // The comments associated with the string.
+ *  comments: Array<string>,
+ *
+ * @param {string} textString
+ * @returns
+ */
+export function* extractStrings(
+    textString: string,
+): Generator<ExtractedString, void, void> {
```

Comment:
That's cool that you can specify the type of arg to `next()`.  It's too bad the types don't default to `void` and `void` for the last two args.